python/docs/source/user_guide/basic_dataframe_operations.rst

```diff
@@ -0,0 +1,169 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+==========================
+Basic DataFrame Operations
+==========================
+
+.. currentmodule:: pyspark.sql
+
+Select
+------
+
+Projects a set of expressions and returns a new :class:`DataFrame`.
+For more detailed usage methods, please refer to :meth:`DataFrame.select`.
+
+Select all columns
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    >>> df = spark.createDataFrame([(2, "Alice"), (5, "Bob")], schema=["age", "name"])
+    >>> df.select('*').show()
+    +---+-----+
+    |age| name|
+    +---+-----+
+    |  2|Alice|
+    |  5|  Bob|
+    +---+-----+
+
+
+Select by column names(string) or expressions (:class:`Column`)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+    >>> df = spark.createDataFrame([(2, "Alice"), (5, "Bob")], schema=["age", "name"])
+    >>> df.select('name', (df.age + 10).alias('age')).show()
```

Comment:
A minor point, but should we mention or demonstrate somehow that you can also select columns using the `__getitem__` syntax (`df["age"]`) in addition to the `__getattr__` syntax (`df.age`) demonstrated here?

Reply:
Okay, I will add an example based on `df["age"]`

Reply:
In fact, I think it's important to use the `__getitem__` syntax as the preferred syntax in these guides, in order to be consistent with the recommendation provided in the "Getting Started" guide: https://github.com/apache/spark/blob/af3a22533ca0a11d91cc70c920d5423a8dac9ee5/docs/sql-getting-started.md?plain=1#L120-L124

Reply:
@panbingkun If the user is interested in selecting all available columns, wouldn't `df.show()` be a better option?